addons/project/res_partner.py

```diff
@@ -36,6 +36,7 @@ def _task_count(self, cr, uid, ids, field_name, arg, context=None):
         'task_count': fields.function(_task_count, string='# Tasks', type='integer'),
     }
 
+    @api.cr_uid_id_context
```

Comment:
Why not just rename the parameter record_id to id (also on subsequent call to super), to have the automatic recognition of old api style methods kick in?